Dfe.Data.SearchPrototype/Web/Controllers/HomeController.cs

```diff
@@ -54,13 +71,56 @@ public async Task<IActionResult> Index(string searchKeyWord)
         {
             return View();
         }
+
         ViewBag.SearchQuery = searchKeyWord;
 
         SearchByKeywordResponse response =
             await _searchByKeywordUseCase.HandleRequest(
-                new SearchByKeywordRequest(searchKeyWord + "*"));
+                new SearchByKeywordRequest(searchKeyword: searchKeyWord + "*"));
+
+        ViewModels.SearchResults viewModel =
+               CreateViewModel(
+                   response.EstablishmentResults,
+                   new EstablishmentFacetsMapperRequest(response.EstablishmentFacetResults));
+
+        return View("Index", viewModel);
+    }
+
+    /// <summary>
+    /// The action method that composes the view model based on the <see cref="SearchRequest"/> posted from for apply
+    /// filters submission which includes the search keyword and selected facet values used to apply filtering.
+    /// </summary>
+    /// <param name="searchRequestViewModel">
+    /// Encapsulates the search keyword and selected facets used to apply filtering.
+    /// </param>
+    /// <returns>
+    /// An IActionResult contract that represents the result of this action method.
+    /// </returns>
+    [HttpPost]
+    public async Task<IActionResult> SearchWithFilters(SearchRequest searchRequestViewModel)
+    {
+        ViewBag.SearchQuery = searchRequestViewModel.SearchKeyword;
+
+        if (searchRequestViewModel.HasSearchKeyWord &&
+            searchRequestViewModel.HasSelectedFacets)
+        {
+            SearchByKeywordResponse response =
+                await _searchByKeywordUseCase.HandleRequest(
+                    new SearchByKeywordRequest(
+                        searchKeyword: searchRequestViewModel.SearchKeyword + "*",
+                        filterRequests: _requestMapper.MapFrom(searchRequestViewModel.SelectedFacets!)));
+
+            ViewModels.SearchResults viewModel = new()
+            {
+
+                SearchItems = _establishmentResultsToEstablishmentsViewModelMapper.MapFrom(response.EstablishmentResults),
+                Facets = _establishmentFacetsToFacetsViewModelMapper.MapFrom(new EstablishmentFacetsMapperRequest(
+                        response.EstablishmentFacetResults, searchRequestViewModel.SelectedFacets))
+            };
```

Comment:
I took the CreateViewModel private method out - was thinking maybe it points to us needing to do it somewhere else